### ISSUE TITLE
sudo processes

### ISSUE THREAD

**Issue by potomak** [maxogden/monu#14]
I'd like to run a process as super user. Is it possible for monu to ask credentials to run processes that need special permissions?

**Comment by max-mapper** [comment]
I actually have no idea how to implement this. I'm thinking if you launch Monu.app from the command line using 'sudo' it will inherit the sudoness for all spawned child processes, but even then i'm not sure that will work.

Ideally there would be some way to e.g. use Keychain to authenticate once and give permission to Monu to always spawn processes as sudo, but I have no idea how to do that or if that is possible

**Comment by potomak** [comment]
Me neither, I'm going to close this issue. Thanks anyway for your help.

**Comment by max-mapper** [comment]
No worries. I actually am gonna reopen just to see if anyone ever comes through with a suggestion of how they might implement this

**Comment by devopsmariocom** [comment]
@potomak You need to setup sudoers in order to allow to escalate rights without passing password.

So just run `sudo visudo` in command line and add line at the end of file

`YourUserName   ALL = (root) /bin/commandToExecute`

Then you can add to monu config `sudo /bin/commandToExecute` and command is executed as super user. For more info see http://osxdaily.com/2014/02/06/add-user-sudoers-file-mac/

**Comment by potomak** [comment]
@elmariofredo I think you need also to add the tag [`NOPASSWD`](https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man5/sudoers.5.html).

**Comment by devopsmariocom** [comment]
@potomak yes you are right, totally forgot about it, so for others correct sudoers line is

`YourUserName   ALL = (root) NOPASSWD: /bin/commandToExecute`

**Comment by shaynem** [comment]
Try this 

https://www.npmjs.com/package/sudo-fn
